source/adapter.screenshare.js

```diff
@@ -0,0 +1,165 @@
+
+      } else { // regular GetUserMediaRequest
+        baseGetUserMedia(constraints, successCb, failureCb);
+      }
+    };
+
+    window.getUserMedia = window.navigator.getUserMedia;
+
+  } else if (window.navigator.webkitGetUserMedia) {
+    baseGetUserMedia = window.navigator.getUserMedia;
+
+    window.navigator.getUserMedia = function (constraints, successCb, failureCb) {
+      constraints = constraints || {};
+
+      if (constraints.video && !!constraints.video.mediaSource) {
+        if (window.webrtcDetectedBrowser !== 'chrome') {
+          throw new Error('Current browser does not support screensharing');
+        }
+
+        var chromeCallback = function(error, sourceId) {
+          if(!error) {
+            constraints.video.mandatory = constraints.video.mandatory || {};
+            constraints.video.mandatory.chromeMediaSource = 'desktop';
+            constraints.video.mandatory.maxWidth = window.screen.width > 1920 ? window.screen.width : 1920;
+            constraints.video.mandatory.maxHeight = window.screen.height > 1080 ? window.screen.height : 1080;
+
+            if (sourceId) {
+              constraints.video.mandatory.chromeMediaSourceId = sourceId;
+            }
+
+            delete constraints.video.mediaSource;
+
+            baseGetUserMedia(constraints, successCb, failureCb);
+
+          } else {
+            if (error === 'permission-denied') {
+              throw new Error('Permission denied for screen retrieval');
+            } else {
+              throw new Error('Failed retrieving selected screen');
+            }
+          }
+        };
+
+        var onIFrameCallback = function (event) {
+          if (!event.data) {
+            return;
+          }
+
+          if (event.data.chromeMediaSourceId) {
+            if (event.data.chromeMediaSourceId === 'PermissionDeniedError') {
+                chromeCallback('permission-denied');
+            } else {
+              chromeCallback(null, event.data.chromeMediaSourceId);
+            }
+          }
+
+          if (event.data.chromeExtensionStatus) {
+            chromeCallback(event.data.chromeExtensionStatus, null);
+          }
+
+          // this event listener is no more needed
+          window.removeEventListener('message', onIFrameCallback);
+        };
+
+        window.addEventListener('message', onIFrameCallback);
+
+        postFrameMessage({
+          captureSourceId: true
+        });
+
+      } else {
+        baseGetUserMedia(constraints, successCb, failureCb);
+      }
+    };
+
+    window.getUserMedia = window.navigator.getUserMedia;
+
+  } else {
+    baseGetUserMedia = window.navigator.getUserMedia;
+
+    window.navigator.getUserMedia = function (constraints, successCb, failureCb) {
+      constraints = constraints || {};
+
+      if (constraints.video && !!constraints.video.mediaSource) {
+        // check if plugin is ready
+        if(AdapterJS.WebRTCPlugin.pluginState === AdapterJS.WebRTCPlugin.PLUGIN_STATES.READY) {
```

Comment:
@serrynaimo These functions are in AJS.
It means that we can either decide that this file is NOT a standalone and needs to be compile with AJS, or create a third file that contains the shared functions.
Given that we already use other function (getUserMedia defined in AJS), I will assume that this file is NOT a standalone